src/bespokelabs/curator/request_processor/openai_online_request_processor.py

```diff
@@ -674,7 +674,15 @@ def num_tokens_consumed_from_request(
             for message in request_json["messages"]:
                 num_tokens += 4  # every message follows <im_start>{role/name}\n{content}<im_end>\n
                 for key, value in message.items():
-                    num_tokens += len(encoding.encode(value))
+                    try:
+                        num_tokens += len(encoding.encode(str(value)))
+                    except TypeError:
+                        logger.warning(
+                            f"Failed to encode value {value} with tiktoken to count tokens, instead assuming a token for every 4 characters."
+                        )
+                        num_tokens += (
+                            len(str(value)) // 4
+                        )  # assume a token for every 4 characters
```

Comment:
num_tokens += len(str(value)) // 4
(one line and no comment since you just explained it above).